### PR TITLE
[docs][java-client] Add sections for interceptor usage

### PR DESCRIPTION
**Pull request by alpreu**
### Motivation

The Pulsar documentation is missing information and usage examples about `ProducerInterceptor` and `ConsumerInterceptor`.

### Modifications

Added sections for both interceptor types to the java-client documentation.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API: (yes / **no**)
  - The schema: (yes / **no** / don't know)
  - The default values of configurations: (yes / **no**)
  - The wire protocol: (yes / **no**)
  - The rest endpoints: (yes / **no**)
  - The admin cli options: (yes / **no**)
  - Anything that affects deployment: (yes / **no** / don't know)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `doc-not-needed` 
(Please explain why)
  
- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Review by codelipenghui — APPROVED** [review]
Great improve for the client documentation.
I always thought we had this doc, 
this feature was developed a long time ago, 
I didn't expect that there is no documentation until now

File: site2/docs/client-libraries-java.md [hunk 1]
```diff
@@ -652,6 +652,43 @@ Producer<byte[]> producer = client.newProducer()
...
+			@Override
+			void onSendAcknowledgement(Producer producer, Message message, MessageId msgId, Throwable exception) {
+			    // user-defined processing logic
+			}
+        })
+        .create();
+
+```
+
+> **Note:** If you are using multiple interceptors, they apply in the order they are passed to the `intercept` method.
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-java.md:690]:
```suggestion
:::note

If you are using multiple interceptors, they apply in the order they are passed to the `intercept` method.

:::
```

Please follow the syntax here: https://docs.google.com/document/d/12De2btkDHQVaqUlHjTqERmroMLKGhHdiC7rEttFTyqc/edit#bookmark=id.jis8plesauvo

**Review by Anonymitaet — COMMENTED** [review]
Thanks for your contribution!

1. Before submitting a doc PR, please [preview your changes locally](https://docs.google.com/document/d/1wszdtMRo6MhKbVaggPK7_bnKaC4TewuT--GWZZxJNGg/edit#). When submitting a PR, please attach the screenshots of your local preview. In this way, you can get your PR reviewed and merged more quickly.

2. Does this change apply to multiple doc versions? If so, please [update them all](https://github.com/apache/pulsar/tree/master/site2/website/versioned_docs).

3. Please update [Pulsar Feature Matrix (Client + Function)](https://docs.google.com/spreadsheets/d/1YHYTkIXR8-Ql103u-IMI18TXLlGStK8uJjDsOOA0T20/edit#gid=1784579914). You can leave a comment on that doc if you do not have edit access.

Thank you! 😊

**Comment by alpreu** [comment]
I just updated the PR accordingly. The feature matrix will be updated as soon as my access request is accepted.

Please find the screenshots of the changes below:
Screenshot 2022-08-30 at 15 29 59
Screenshot 2022-08-30 at 15 30 17
Screenshot 2022-08-30 at 15 30 27

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**